packages/playwright-core/src/utils/isomorphic/urlMatch.ts

```diff
@@ -51,13 +51,14 @@ export function globToRegex(glob: string): RegExp {
 
     switch (c) {
       case '?':
-        tokens.push('.');
+        // Unescaped ? is treated same as escaped.
+        tokens.push('\\?');
         break;
       case '[':
-        tokens.push('[');
+        tokens.push('\\[');
         break;
       case ']':
-        tokens.push(']');
+        tokens.push('\\]');
```

Comment:
Looks like `?`,`[`,`]` can be passed through to the `default:` case.

Reply:
You are right, updated.